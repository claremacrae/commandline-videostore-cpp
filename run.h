#ifndef COMMANDLINE_VIDEOSTORE_CPP_RUN_H
#define COMMANDLINE_VIDEOSTORE_CPP_RUN_H
#include <iosfwd>
#include <string>

void run(std::istream& in, std::ostream& out);

void calculateAmountAndPoints(const std::string& category,
                              int daysRented,
                              double& thisAmount,
                              int& frequentRenterPoints);

#endif
