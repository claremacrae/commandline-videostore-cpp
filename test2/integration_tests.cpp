#include "catch2/catch.hpp"
#include "ApprovalTests.hpp"

#include "run.h"

using namespace ApprovalTests;

TEST_CASE("empty input") {
  std::stringstream input;
  std::stringstream output;
  run(input, output);
  Approvals::verify(output.str());
}

TEST_CASE("real input") {
  std::string typed = R"(Clare
0 1
0 2
0 3
0 4
9 1
9 2
9 3
9 4
12 1
12 2
12 3
12 4

)";
  std::stringstream input;
  input << typed;
  std::stringstream output;
  run(input, output);
  Approvals::verify(output.str());
}
