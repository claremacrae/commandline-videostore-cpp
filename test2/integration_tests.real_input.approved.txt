0: The Shawshank Redemption (1994)
1: The Godfather (1972)
2: The Godfather: Part II (1974)
3: The Dark Knight (2008)
4: Pulp Fiction (1994)
5: The Good, the Bad and the Ugly (1966)
6: The Lord of the Rings: The Return of the King (2003)
7: Fight Club (1999)
8: The Lord of the Rings: The Fellowship of the Ring (2001)
9: Interstellar (2014)
10: Whiplash (2014)
11: Birdman (2014)
12: Up (2009)
13: WALLÂ·E (2008)
Enter customer name: Choose movie by number followed by rental days, just ENTER for bill:
Rental Record for Clare
	The Shawshank Redemption (1994)	2.0
	The Shawshank Redemption (1994)	2.0
	The Shawshank Redemption (1994)	3.5
	The Shawshank Redemption (1994)	5.0
	Interstellar (2014)	3.0
	Interstellar (2014)	6.0
	Interstellar (2014)	9.0
	Interstellar (2014)	12.0
	Up (2009)	1.5
	Up (2009)	1.5
	Up (2009)	1.5
	Up (2009)	3.0
You owed 50.0
You earned 15 frequent renter points

